docs/user_guide/reductions.rst

```diff
@@ -0,0 +1,263 @@
+The Reductions Approach to Disparity Mitigation
+===============================================
+
+The "Reductions Approach" is named because it *reduces* the problem of training a
+classifier while satisfying a disparity constraint to that of training a classifier
+on weighted data.
+This technique was first described in [#1]_.
```

Comment:
@riedgar-ms If I'm following you, the intention here to explain the internals of the algorithm to someone is interested?

If so I can help merge in some of the discussion from other issues (eg, https://github.com/fairlearn/fairlearn/issues/444, https://github.com/fairlearn/fairlearn/issues/460 and https://github.com/fairlearn/fairlearn/issues/439) or help with reviewing if that's helpful.  I can bring a helpful lack of experience and knowledge to the task :)

Reply:
Yes - especially when it comes to writing new Moments. There's also the secondary goal of explaining how things work to myself, since I've been getting behind on that bit. However, since @romanlutz  also has some changes in flight, this PR is on hold right now.

Reply:
ah okay then I will ignore in the meantime, thanks! 👍

Reply:
@kevinrobinson if you want to take a look again, I think this is ready for review.

Reply:
@riedgar-ms Thanks for the ping, it's awesome to see all the work towards this!  👍 I can't dig into this now but here's three quick thoughts.  I can review afterward too if you all want to merge now.

1. Describing the internals of how the algorithm works for binary classification seems helpful for the team, especially for some of the conversations linked here.  This seems particularly helpful for team members who are trying to make improvements or changes to this algorithmic approach, or to evaluate it or compare it to other algorithmic approaches.  I'm interested and curious in this, so would love to read through and check my understanding and help with revising.  We can also just ship and I can do that after, I don't want to hold up progress.
2. I'm skeptical this document will be helpful to anyone outside of algorithmic ML fairness researchers, and I don't think this is an important audience for the project at this point.  For data science practioners who discover the project, I'm assuming some immediate needs will be: evaluating whether they can use this algorithm for their work, understanding situations where it can best help with enforcing constraints, and learning what major design choices this approach provides to them as users.  I think that work would be super valuable to the project, but would look really different - it might start with calling out non-determistic predictions, focusing on evaluation with train/validation sets, etc.  I get that addressing that was not your intention in this pull request, but I'm calling it out since it seems like valuable work we could be doing going forward as a next step.
3. My understanding is that binary classification is rarely used by practitioners.  That being said, I think it's super valuable to be working on user guides and documentation, so if this is where it's easiest to get started let's do it! :)

Thanks, I'm looking forward to reviewing and using this to check my understanding, and sorry with the timing that I can't do that for another week or so.  I have no objections to reviewing after if you want to 🚢  in the meantime 👍